c++11/include/opentracing/propagation.h

```diff
@@ -0,0 +1,139 @@
+#ifndef OPENTRACING_PROPAGATION_H
+#define OPENTRACING_PROPAGATION_H
+
+#include <opentracing/preprocessor.h>
+#include <opentracing/stringref.h>
+#include <opentracing/util.h>
+#include <functional>
+#include <string>
+
+namespace opentracing {
+inline namespace OPENTRACING_VERSION_NAMESPACE {
+enum class SpanReferenceType {
+  // ChildOfRef refers to a parent Span that caused *and* somehow depends
+  // upon the new child Span. Often (but not always), the parent Span cannot
+  // finish unitl the child Span does.
+  //
+  // An timing diagram for a ChildOfRef that's blocked on the new Span:
+  //
+  //     [-Parent Span---------]
+  //          [-Child Span----]
+  //
+  // See http://opentracing.io/spec/
+  //
+  // See opentracing.ChildOf()
+  ChildOfRef = 1,
+
+  // FollowsFromRef refers to a parent Span that does not depend in any way
+  // on the result of the new child Span. For instance, one might use
+  // FollowsFromRefs to describe pipeline stages separated by queues,
+  // or a fire-and-forget cache insert at the tail end of a web request.
+  //
+  // A FollowsFromRef Span is part of the same logical trace as the new Span:
+  // i.e., the new Span is somehow caused by the work of its FollowsFromRef.
+  //
+  // All of the following could be valid timing diagrams for children that
+  // "FollowFrom" a parent.
+  //
+  //     [-Parent Span-]  [-Child Span-]
+  //
+  //
+  //     [-Parent Span--]
+  //      [-Child Span-]
+  //
+  //
+  //     [-Parent Span-]
+  //                 [-Child Span-]
+  //
+  // See http://opentracing.io/spec/
+  //
+  // See opentracing.FollowsFrom()
+  FollowsFromRef = 2
+};
+
+// Carrier format values.
+enum class CarrierFormat {
```

Comment:
Is this the right type to use to specify the carrier format? It's not clear to me how something like LightStep's [custom binary format](https://github.com/lightstep/lightstep-tracer-cpp/blob/master/src/c%2B%2B11/lightstep/propagation.h#L176) or other custom carriers would fit into this.